test/stake.test.ts

```diff
@@ -201,4 +244,95 @@ describe('StakingPool.stake', () => {
+      await testEnv.rewardAsset.connect(deployer).faucet();
+      await testEnv.rewardAsset.connect(deployer).approve(testEnv.stakingPool.address, RAY);
+      await testEnv.stakingPool
+        .connect(deployer)
+        .initNewPool(rewardPersecond, startTimestamp, duration);
+      await testEnv.stakingAsset.connect(alice).faucet();
+      const tx = await testEnv.stakingAsset.connect(alice).approve(testEnv.stakingPool.address, RAY);
+      await advanceTimeTo(await getTimestamp(tx), startTimestamp);
+      await testEnv.stakingPool.connect(alice).stake(stakeAmount);
+
+    });
+    
+    it('rewardPerSecond is changed and stake in pool', async () => {
+      const tx = await testEnv.stakingPool.connect(deployer).inputNextReward(inputAmount); 
+
+      const poolDataBefore = await getPoolData(testEnv);
+      const userDataBefore = await getUserData(testEnv, alice);
+      await advanceTimeTo(await getTimestamp(tx), passTimestamp);
+
+      const [expectedPoolData_1, expectedUserData_1] = expectDataAfterStake(
+        poolDataBefore,
+        userDataBefore,
+        passTimestamp,
+        BigNumber.from(0)
+      );
+      
+      const stakeTx = await testEnv.stakingPool.connect(alice).stake(stakeAmount);
+      
+      const [expectedPoolData_2, expectedUserData_2] = expectDataAfterStakeSetReward(
+        expectedPoolData_1,
+        expectedUserData_1,
+        await getTimestamp(stakeTx),
+        stakeAmount,
+        nextRewardPersecond,
+        duration
+      );
+      
+      const poolDataAfter = await getPoolData(testEnv);
+      const userDataAfter = await getUserData(testEnv, alice);
+
+      expect(poolDataAfter).to.be.equalPoolData(expectedPoolData_2);
+      expect(userDataAfter).to.be.equalUserData(expectedUserData_2);
+      
+    });
+
+    it('rewardPerSecond is changed and stake in pool twice', async () => {
+      const tx = await testEnv.stakingPool.connect(deployer).inputNextReward(inputAmount); 
+
+      const poolDataBefore = await getPoolData(testEnv);
+      const userDataBefore = await getUserData(testEnv, alice);
+      await advanceTimeTo(await getTimestamp(tx), passTimestamp);
+
+      const [expectedPoolData_1, expectedUserData_1] = expectDataAfterStake(
+        poolDataBefore,
+        userDataBefore,
+        passTimestamp,
+        BigNumber.from(0)
+      );
+      
+      const stakeTx_1 = await testEnv.stakingPool.connect(alice).stake(stakeAmount);
+      const [expectedPoolData_2, expectedUserData_2] = expectDataAfterStakeSetReward(
+        expectedPoolData_1,
+        expectedUserData_1,
+        await getTimestamp(stakeTx_1),
+        stakeAmount,
+        nextRewardPersecond,
+        duration
+      );
+      
+      await advanceTimeTo(await getTimestamp(stakeTx_1), endTimestamp);
+      const stakeTx_2 = await testEnv.stakingPool.connect(alice).stake(stakeAmount);
+
+      const [expectedPoolData_3, expectedUserData_3] = expectDataAfterStake(
+        expectedPoolData_2,
```

Comment:
After these `expectedPoolData_` parameter issues are solved, I'd review the rest code and logic.